jina/drivers/rank/aggregate/__init__.py

```diff
@@ -72,30 +72,35 @@ def _insert_query_matches(
 
     @staticmethod
     def _group_by(match_idx, col_name):
-        # sort by ``col
+        """
+        Create an list of numpy arrays with the same ``col_name`` in each position of the list
+
+        :param match_idx: Numpy array of Tuples with document id and score
+        :param col_name:  Column name in the structured numpy array of Tuples
+
+        :return: List of numpy arrays with the same ``doc_id`` in each position of the list
+        :rtype: np.ndarray.
+        """
         _sorted_m = np.sort(match_idx, order=col_name)
-        _, _doc_counts = np.unique(_sorted_m[col_name], return_counts=True)
-        # group by ``col``
-        return np.split(_sorted_m, np.cumsum(_doc_counts))[:-1]
+        list_numpy_arrays = []
```

Comment:
Just curious, why do u think this is an optimization?

Reply:
I think it's because` _, _doc_counts = np.unique(_sorted_m[col_name], return_counts=True)` will do the sorting again, so at least O(long)?